packages/js-drive/lib/abci/handlers/processProposalHandlerFactory.js

```diff
@@ -13,17 +13,96 @@ const proposalStatus = {
 };
 
 /**
- *
+ * @param {deliverTx} deliverTx
+ * @param {BaseLogger} logger
+ * @param {GroveDBStore} groveDBStore
+ * @param {BlockExecutionContext} blockExecutionContext
+ * @param {beginBlock} beginBlock
+ * @param {endBlock} endBlock
+ * @param {updateConsensusParams} updateConsensusParams
+ * @param {rotateValidators} rotateValidators
  * @return {processProposalHandler}
  */
-function processProposalHandlerFactory() {
+function processProposalHandlerFactory(
+  deliverTx,
+  logger,
+  groveDBStore,
+  blockExecutionContext,
+  beginBlock,
+  endBlock,
+  updateConsensusParams,
+  rotateValidators,
+) {
   /**
    * @typedef processProposalHandler
    * @return {Promise<abci.ResponseProcessProposal>}
    */
-  async function processProposalHandler() {
+  async function processProposalHandler({
+    height,
+    txs,
+    coreChainLockedHeight,
+    version,
+    proposedLastCommit: lastCommitInfo,
+    time,
+    proposerProTxHash,
+  }) {
+    const consensusLogger = logger.child({
+      height: height.toString(),
+      abciMethod: 'processProposal',
+    });
+
+    await beginBlock(
+      {
+        lastCommitInfo,
+        height,
+        coreChainLockedHeight,
+        version,
+        time,
+        proposerProTxHash: Buffer.from(proposerProTxHash),
+      },
+      consensusLogger,
+    );
+
+    const txResults = [];
+
+    for (const tx of txs) {
+      txResults.push(await deliverTx(tx, consensusLogger));
```

Comment:
You don't need valid and invalid counter in block execution context. You can do it right here